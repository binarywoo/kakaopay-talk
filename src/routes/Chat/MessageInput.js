import React, { useCallback } from 'react'
import {
  Avatar,
  Upload,
  Row,
  Col,
  Input,
  Icon,
  message as antdMessage
} from 'antd'
import classNames from 'classnames'
import MessageModal from '../../components/MessageModal/index'

const MessageInput = ({
  message,
  onChange,
  onSendMessage,
  onClickInvitation,
  sendImageMessage,
  showHiddenMenu,
  setShowHiddenMenu
}) => {
  const beforeImageUpload = useCallback(file => {
    setShowHiddenMenu(false)
    const hide = antdMessage.loading('이미지 업로드 중..', 0)
    sendImageMessage(file).then(() => hide())
    return false
  }, [])

  const onClickNotReady = useCallback(() => {
    MessageModal({
      type: 'info',
      title: '준비중',
      content: '준비중인 기능입니다.'
    })
  }, [])

  return (
    <div className='chat-input-wrapper' offsetBottom={0}>
      <div
        style={{
          padding: 12,
          backgroundColor: '#fff',
          borderTop: '1px solid #bbbbbb'
        }}
      >
        <Input.Search
          value={message}
          onChange={onChange}
          addonBefore={
            <Icon
              type={showHiddenMenu ? 'minus' : 'plus'}
              onClick={() => setShowHiddenMenu(!showHiddenMenu)}
            />
          }
          enterButton={<Icon type='arrow-up' />}
          onSearch={message.length > 0 ? onSendMessage : null}
        />
      </div>
      <div
        className={classNames(
          'hidden-menu',
          showHiddenMenu ? 'show' : 'hidden'
        )}
      >
        <Row gutter={12}>
          <Col span={6}>
            <Avatar
              className='invitation'
              icon='user-add'
              onClick={onClickInvitation}
            />
            초대
          </Col>
          <Col span={6}>
            <Upload
              key='picture'
              accept='image/*'
              beforeUpload={beforeImageUpload}
            >
              <Avatar className='image' icon='picture' />
              이미지
            </Upload>
          </Col>
          <Col span={6}>
            <Avatar className='pay' icon='wallet' onClick={onClickNotReady} />
            송금
          </Col>
          <Col span={6}>
            <Avatar
              className='map'
              icon='environment'
              onClick={onClickNotReady}
            />
            지도
          </Col>
        </Row>
      </div>
    </div>
  )
}

export default MessageInput
