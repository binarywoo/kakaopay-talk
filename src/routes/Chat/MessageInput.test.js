import React from 'react'
import { shallow } from 'enzyme'
import { Input } from 'antd'
import MessageInput from './MessageInput'

describe('MessageInput', () => {
  let wrapper = null

  const props = {
    message: '',
    onChange: () => null,
    onSendMessage: () => null,
    onClickInvitation: () => null,
    sendImageMessage: () => null,
    showHiddenMenu: false,
    setShowHiddenMenu: () => null
  }

  it('정상적으로 렌더링이 된다.', () => {
    wrapper = shallow(<MessageInput {...props} />)
  })

  it('스냅샷과 매칭이 된다.', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('message를 정상적으로 렌더링한다.', () => {
    const newProps = Object.assign({}, props, {
      message: 'test'
    })
    wrapper = shallow(<MessageInput {...newProps} />)
    expect(wrapper.find(Input.Search).prop('value')).toBe('test')
  })

  it('히든메뉴를 정상적으로 보여준다.', () => {
    expect(wrapper.find('.hidden-menu').hasClass('hidden')).toBe(true)
    const newProps = Object.assign({}, props, {
      showHiddenMenu: true
    })
    wrapper = shallow(<MessageInput {...newProps} />)
    expect(wrapper.find('.hidden-menu').hasClass('show')).toBe(true)
  })
})
