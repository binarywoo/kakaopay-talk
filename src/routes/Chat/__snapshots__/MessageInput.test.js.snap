// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageInput 스냅샷과 매칭이 된다. 1`] = `
<div
  className="chat-input-wrapper"
  offsetBottom={0}
>
  <div
    style={
      Object {
        "backgroundColor": "#fff",
        "borderTop": "1px solid #bbbbbb",
        "padding": 12,
      }
    }
  >
    <Search
      addonBefore={
        <Icon
          onClick={[Function]}
          type="plus"
        />
      }
      enterButton={
        <Icon
          type="arrow-up"
        />
      }
      onChange={[Function]}
      onFocus={[Function]}
      onSearch={null}
      value=""
    />
  </div>
  <div
    className="hidden-menu hidden"
  >
    <Row
      gutter={12}
    >
      <Col
        span={6}
      >
        <Avatar
          className="invitation"
          icon="user-add"
          onClick={[Function]}
          shape="circle"
          size="default"
        />
        초대
      </Col>
      <Col
        span={6}
      >
        <Upload
          accept="image/*"
          action=""
          beforeUpload={[Function]}
          className=""
          data={Object {}}
          disabled={false}
          key="picture"
          listType="text"
          multiple={false}
          showUploadList={true}
          supportServerRender={true}
          type="select"
        >
          <Avatar
            className="image"
            icon="picture"
            shape="circle"
            size="default"
          />
          이미지
        </Upload>
      </Col>
      <Col
        span={6}
      >
        <Avatar
          className="pay"
          icon="wallet"
          onClick={[Function]}
          shape="circle"
          size="default"
        />
        송금
      </Col>
      <Col
        span={6}
      >
        <Avatar
          className="map"
          icon="environment"
          onClick={[Function]}
          shape="circle"
          size="default"
        />
        지도
      </Col>
    </Row>
  </div>
</div>
`;
